### ISSUE TITLE
Plan 2021-09 release

### ISSUE THREAD

**Issue by cgolubi1** [buttonmen-dev/buttonmen#2733]
Do all testing and deployments for the 2021-09 buttonweavers.com prod release.

Major changes expected to be in this release:
* #2732 - player info migration from 'gender' to 'pronouns'
* #2389 - allow custom recipes on dev sites (should be a noop on staging and prod)

The plan is to cut this release, even though there's not very much player-visible in it, because we want the under-the-hood changes for custom recipes to go out on their own so we can catch any regressions or weirdness.

If there is anything very important you expect to see in this release and don't, please complain quickly.

**Comment by cgolubi1** [comment]
If folks can poke around on the staging site a little bit, that'd be great --- as you can see from #2751 basically the only visible change is the pronouns change.  Other changes are the backend for custom recipes, and a minor change to move some more things to BMDB.  I'm really just looking for a smoke test in which people poke around on staging for a bit and i make sure no weird errors show up in the logs.

Apologies again for not having gotten this done in June.

**Comment by dwvanstone** [comment]
I've poked around the site - added gender, removed gender, created a game, poked around the forum & history. Nothing weird happened.

**Comment by cgolubi1** [comment]
@blackshadowshade --- i created a CustomBM game on staging using the API - https://staging.buttonweavers.com/ui/game.html?game=216

I know our plan is to eventually allow CustomBM games on staging.  However, the only actual logic in your actual code which is supposed to restrict this is https://github.com/buttonmen-dev/buttonmen/blob/master/src/engine/BMInterfaceGame.php#L249-L252 --- that looks for dev sites.

It seems like the problem is that staging has inconsistent site types:
* The UI's Config.js contains: `Config.siteType = 'staging';`
* The database's config table contains: `| site_type | development |`

I went ahead and updated the database to contain site_type = staging, and after i did that, i got the expected failure:

```
>>> r = b.create_game('CustomBM', custom_recipe_array=['(1) (1) (1)'], opponent='blackshadowshade', obutton='Agatha')
>>> r.status
u'failed'
>>> r.message
u'Custom recipes can only be used on development sites.'
```

Does that seem all good, or should we worry about anything else here?

**Comment by blackshadowshade** [comment]
Ah, the inconsistent site type thing again. I believe that we thought about fixing that inconsistency by tweaking the setup scripts somehow.

That behaviour looks fine after the change, yes.

**Comment by cgolubi1** [comment]
I want us to deploy a fix for #2757 before wrapping up this merge freeze period.

**Comment by cgolubi1** [comment]
Okay, prod is updated.  I was so keen to get a fix out because the symptom of the bug wasn't only that the site gave API messages i didn't like to players who tried to create games with invalid CustomBM recipes; it was that when someone created a game with a *valid* CustomBM recipe, an unplayable game actually got created on the site.

Specifically, https://www.buttonweavers.com/ui/game.html?game=77312 is the example i created.  It was an open game, and someone tried to join it, yielding an internal error:

```
Caught exception in BMInterface::select_button: Infinite loop detected when advancing game state. Final game state: SPECIFY_RECIPES, referer: https://www.buttonweavers.com/ui/open_games.html
```

Anyway, i think that game is now in some sort of harmless "unstarted" state --- not in open games any more; i'll find out if the other player is seeing any sort of issue with it, and we can move it to a more dedicated "broken" state if so, and not worry about it if not.  But i didn't want players to stack up a bunch of games in that state before we could close the hole.
